packages/flutter/lib/src/widgets/icon.dart

```diff
@@ -122,6 +128,18 @@ class Icon extends StatelessWidget {
     );
   }
 
+  /// Wraps the widget with a Semantics widget if [label] is set.
+  Widget wrapWithSemantics(Widget widget) {
+    if (label == null) {
+      return new ExcludeSemantics(child: widget);
+    } else {
+      return new Semantics(
+        child: widget,
```

Comment:
I would expect that in this case you'll also have to ExcludeSemantics on the widget, no?